### PR TITLE
Make ES|QL tests compatible with stateless

### PR DESCRIPTION
**Pull request by mark-vieira**
This PR makes some changes to the mixed-cluster and multi-node ES|QL tests to make them compatible with stateless environments. The main change here is removing the test index settings in `FieldExtractorTestCase`. Setting replicas and shards isn't applicable in stateless, and in fact, setting replicas to 0 results in shard availability errors. Removing these settings _should_ be benign in stateful, as we'll just use the defaults.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Review by luigidellaquila — APPROVED** [review]
LGTM, thanks!

File: x-pack/plugin/esql/qa/server/multi-node/src/yamlRestTest/java/org/elasticsearch/xpack/esql/qa/multi_node/EsqlClientYamlIT.java [hunk 1]
```diff
@@ -20,12 +19,7 @@
 
 public class EsqlClientYamlIT extends ESClientYamlSuiteTestCase {
     @ClassRule
-    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
-        .distribution(DistributionType.DEFAULT)
-        .nodes(2)
-        .setting("xpack.security.enabled", "false")
-        .setting("xpack.license.self_generated.type", "trial")
-        .build();
+    public static ElasticsearchCluster cluster = Clusters.testCluster();
```
Comment (bpintea) [inline x-pack/plugin/esql/qa/server/multi-node/src/yamlRestTest/java/org/elasticsearch/xpack/esql/qa/multi_node/EsqlClientYamlIT.java:22]: 👍

**Review by bpintea — APPROVED** [review]
LGTM

**Pull request merged**